### ISSUE TITLE
[Question] Face tracking

### ISSUE THREAD

**Issue by G3z** [HippoAR/react-native-arkit#154]
I know ARKit has face tracking
https://developer.apple.com/videos/play/fall2017/601/
I was wondering if you plan to implement such a feature or you decided not to (and why)

**Comment by macrozone** [comment]
@G3z I don't have an iphoneX to try it out, but there is nothing that would prevent an implementation. PRs would be welcome!

It looks not so complicated: https://developer.apple.com/documentation/arkit/arfacetrackingconfiguration

I will soon implement the newer Anchor types like Image or vertical planes (coming with arkit 1.5 on current ios beta), after that its probably easier to add support for ARFaceAnchor

**Comment by macrozone** [comment]
@G3z I am  adding new anchor-types and it should be easier now to add facetracking: https://github.com/HippoAR/react-native-arkit/pull/155/files#diff-f21b481ab0f573fb47650585ffa63fe7R510

We probably just need to add a new property to the `<ARKit />`-component to enable facetracking

**Comment by evezzaro** [comment]
Any news on this?

**Comment by macrozone** [comment]
@evezzaro it's not planned for the moment from my side. PR's are highly encouraged!

**Comment by RWOverdijk** [comment]
I'm using MLKit for this now... But it would be really cool if this did work. A lot faster, too 😄

**Comment by HappyCodingLover** [comment]
@RWOverdijk hi!
I am curious that we can make AR face mask just like Instagram, with this awesome library.
Or have you already done with something else?

**Comment by RWOverdijk** [comment]
I solved it by building the app using swift.

**Comment by HappyCodingLover** [comment]
thanks for reply, @RWOverdijk 
do you think we have possibility on react native?
cause my app is written in RN currently and just is wanting to add, AR - face mask function 😊

**Comment by RWOverdijk** [comment]
I don't know haha. It'd probably have to be built. Maybe someone else has an answer! :)

**Comment by HappyCodingLover** [comment]
thanks, @RWOverdijk

**Comment by macrozone** [comment]
unfortunatly, this package is no longer maintained. I know that, because i maintained it ;-) 

Problem was that i had no project to use it and android / arcore version never happened (or it did happen, but they decided to fork did not merge into this package here).

you might try https://github.com/viromedia/viro/ which supports both ios and android, but it also staled a bit.

i think the AR on react-native chapter is not closed yet, but currently, its in limbo.

I would love if someone would pick up the work here or on viro. It's actually not that hard.

**Comment by HappyCodingLover** [comment]
thanks for information, it was oasis for me. @macrozone 
so, what i want to archieve is the AR-face mask functionality just like one in Snapshot or Instagram,
I see many stars on this project( 1.6K ), but not sure my goal is doable with this library.
any answer for me?

**Comment by macrozone** [comment]
@HappyCodingLover the library here received its last update 3 years ago. that time, Arkit had no support yet for face recognition or similar. So the answer is no, unfortunatly

**Comment by macrozone** [comment]
@HappyCodingLover having said that, i did not read my own comment from 2018...

**Comment by HappyCodingLover** [comment]
@macrozone clear, so now, my possible solution should be found in Viro
do u know if Viro supports face mask?

**Comment by RWOverdijk** [comment]
@HappyCodingLover I would ask it on that repo instead. Or try building your own! (it really is fun and there are plenty of tutorials for the native side of things).

**Comment by HappyCodingLover** [comment]
@RWOverdijk yeap, I will try and let you know if I make something cool.
thanks for your guidance

**Comment by br-matt** [comment]
Hey @macrozone there are some pretty serious PRs open and waiting review ( worldmap sharing, some rotation helpers and others ) 

Any way you would think about giving myself and @JackSoby collaborator to start bringing some of those in ? I think people would add things like face detection if not such big PRs just sitting there.

**Comment by code-matt** [comment]
Hey @macrozone there are some pretty serious PRs open and waiting review ( worldmap sharing, some rotation helpers and others )

Any way you would think about giving myself and @JackSoby collaborator to start bringing some of those in ? I think people would add things like face detection if not such big PRs just sitting there.

**Comment by macrozone** [comment]
@code-matt i added you as admin to the repo!

i am unsure who has admin right on the npm repo though, let' me have a quick look...

**Comment by macrozone** [comment]
i am currently unsue who has admin right on the npm organisation, unfortunatly, its not me, i am just member. I am not sure if i can reach the other contributors.

@gtfargo you seem to be member of hippAr. current admin on npm are zehaoli and jacy, can you reach these guyes?

@code-matt in the worst case you might need to create a new package in npm, but i guess that is not that dramatic. Make sure that you have enough people with admin rights there...

**Comment by code-matt** [comment]
@macrozone  Okay, will just create a new one if it comes down to that.

I am not seeing the ability to merge requests still though. Not that I would now, need to look some of the recent PRs over with @JackSoby

**Comment by macrozone** [comment]
@code-matt you have to accept the invitation probably, thats sometimes a bit hidden in github

**Comment by code-matt** [comment]
@macrozone  Ah sorry, ancient email : ) will do

**Comment by danbachar** [comment]
Hey @code-matt @macrozone @JackSoby, sorry for bumping this again, but I really want to use this library and have currently no idea what‘s the status. Is this package maintained, was it forked, and what eventually happened with RN AR face tracking?
Thanks a bunch,
Dan

**Comment by macrozone** [comment]
hi @danbachar no, its not maintained, maybe checkout https://github.com/ViroCommunity/viro/ which is still activily maintained

**Comment by JackSoby** [comment]
@danbachar If you have a PR you would like to open I would be happy to look at it

**Comment by danbachar** [comment]
Thanks! Will do.

**Comment by emphaticaditya** [comment]
does RN have ARKit face tracking support?

**Comment by JackSoby** [comment]
> does RN have ARKit face tracking support?

I Dont believe so, but if you would like to open up a PR I would be happy to approve!
